### DISCUSSION TITLE
Daily trip odometer

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#376] [Q&A] [answered]
For at least twenty years, I have had a habit of resetting the trip odometer after every fill-up. Simple and effective, but it "blocks" this counter for other uses (and sometimes it would be useful)...

Maybe we could have a panel with mileage since last refueling (for example) which can be reset at any time? Perhaps it would be good to have the option to manually correct it, in case we forgot to reset it at the gas station, or drove some distance without Cruiser. We could use long click and double click to operate it to prevent accidental action.

**Comment by devemux86** [comment]
@Ciutnik thanks for the idea, something like this is possible.

Cruiser panels are very versatile, the possibilities are endless.
e.g. OsmAnd even provides weather and sunset / sunrise info. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
This is not a very important function, but since I do not have a fuel level indicator, and I only estimate it according to mileage (for me it is 220–230 km to the reserve), I would be happy to use it 🙂

I imagine it would be possible to define a value after which the panel would turn red (like the battery level)❗

**Comment by Ciutnik** [comment]
Let me come back to the idea of a daily trip counter for a moment. 
A short press gives you the option to reset - great! Could you please use a long press to set the initial value (and maybe even negative values, in order to count the mileage as if "from the end")? This would be a nice addition to this counter.

Today I once again forgot to reset the counter while refueling and did it 10 km away. Now I have to remember about this. And all I had to do was enter "10" and I wouldn't have to burden my unreliable memory 🙂

**Reply by MotoUKRider** [reply]
@Ciutnik Oh for the days of having to open the fuel-cap and rock the bike from side-to-side and guestimate how much you had left... If we were super-clever, we'd also remember *not* to ride on the "reserve" tap :fuelpump:

**Reply by devemux86** [reply]
@Ciutnik thanks for the suggestion, we could improve the odometer menu with actions like:
- Reset
- Set

**Reply by Ciutnik** [reply]
@MotoUKRider 
> Oh for the days of having to open the fuel-cap and rock the bike from side-to-side

For me, such days are STILL an everyday occurrence 🙂 But I also remember the days when I rode a motorbike with a sidecar. Estimating the fuel level in the tank was even more challenging 😀 Incidentally, driving it itself was also a challenge. A fun challenge.

**Reply by DolomitesTom** [reply]
good idea... was also a problem with my RdGA tour. Had to change the device for some sections because I do not have wireless loading facility with all my devices... and then have no full counter.

**Comment by ToToGSARider** [comment]
wie setzt man den Tageskilometerzähler zum denn wieder auf Null zurück

How do you reset the trip odometer to zero again?

**Reply by devemux86** [reply]
@ToToGSARider 

Tap the panel.

Tippen Sie auf das Bedienfeld.

**Comment by devemux86** [comment]
Coming in next Cruiser:

<img src="https://github.com/user-attachments/assets/3fa7deef-19dc-46d8-9375-0ba54ebd16e2" width="50%"/>

**Comment by devemux86** [comment]
Trip meter new menu with "Reset" and "Set" is now available:

- https://github.com/devemux86/cruiser/discussions/610

**Comment by DolomitesTom** [comment]
will this be only available in Cruiser 5 and never Cruiser 4? I have an issue with the daily trip odometer. I have it only visible when looking at the map, not navigation and not follow mode. But I would nevertheless like to count, but this is not working. It only counts the trip kilometers when it is visible during navigation. Can it also count in the background and when I stop navigation and look at the map, then I would like to see both.

**Reply by devemux86** [reply]
> will this be only available in Cruiser 5 and never Cruiser 4?

Development now continues in Cruiser 5.

(I can only maintain 1 map engine) :slightly_smiling_face:

**Comment by devemux86** [comment]
> I have it only visible when looking at the map, not navigation and not follow mode. But I would nevertheless like to count, but this is not working. It only counts the trip kilometers when it is visible during navigation. Can it also count in the background and when I stop navigation and look at the map, then I would like to see both.

This is how we discussed the trip meter to work, only during movement in navigation or follow location.
- Discussion is probably here: https://github.com/devemux86/cruiser/discussions/494

I think it works similarly to the average speed. We could change this behavior if it is needed.

**Reply by bios4** [reply]
> This is how we discussed the trip meter to work, only during movement in navigation or follow location.

This is perfectly fine, I think the question is if it is possible to implement it like this:

- COUNT only during navigation/follow (without the panel being visible)
- DISPLAY the panel only during planning/map browsing (then showing the values from navigation/follow)

**Reply by devemux86** [reply]
> DISPLAY the panel only during planning/map browsing (then showing the values from navigation/follow)

Display of panels is available as option in all modes: navigation, follow location, browse map.

> COUNT only during navigation/follow (without the panel being visible)

Always counting automatically even without being visible is a bit confusing and dangerous.
As it then modifies the recorded data and users will complain that their records are changed.

**Comment by Ciutnik** [comment]
Since I only just noticed it, it's probably not important: but shouldn't the odometers be independent for each profile? 

I just accidentally recorded kilometers on my motorcycle's odometer while car driving 🙂

**Reply by devemux86** [reply]
@Ciutnik thanks for the report.

Indeed and the same happens with the average speed.
All will work properly and separately in the next update.

**Reply by Ciutnik** [reply]
> All will work properly and separately in the next update.

Great! Thank you very much 🙂

**Comment by devemux86** [comment, marked as answer]
- Trip meter new menu with "Reset" and "Set" is available in https://github.com/devemux86/cruiser/discussions/641
